managers/views.py

```diff
@@ -89,17 +103,31 @@ def get_context_data(self, **kwargs: Any) -> dict:
         return context_data
 
     def get_success_url(self) -> str:
+        logger.info(f"User: {self.request.user} is in ProjectUpdate view for project with id: {self.kwargs['pk']}")
         return reverse("custom-project-detail", kwargs={"pk": self.kwargs["pk"]})
 
+    def form_valid(self, form: ProjectForm) -> HttpRequest:
+        project = form.save()
+        logger.info(f"Project: {project} with id: {project.pk} has been updated by user: {self.request.user}")
+        return super().form_valid(form)
+
 
 class ProjectDeleteView(DeleteView):
     model = Project
 
     def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
         if not request.user.is_admin:
+            logger.debug(f"User: {request.user} want to delete project")
             return redirect(reverse("home"))
 
         return super().dispatch(request, *args, **kwargs)
 
     def get_success_url(self) -> str:
         return reverse("custom-projects-list")
+
+    def delete(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
+        project = self.get_object()
```

Comment:
User `super()` here too instead of rewriting whole method.

Reply:
I am not sure if I did it in good way,
at the first moment i just wrote it like this
```python
    def delete(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpResponse:
        super().delete(request, args, kwargs)
        success_url = self.get_success_url()
        logger.info(f"Project with id: {request.project.pk} has been deleted by user with id: {self.request.user.pk}")
        return HttpResponseRedirect(success_url)
```
But I didn't choose this way because I wasn't sure that it is proper way.